cmd/machine-api-operator/start.go

```diff
@@ -101,3 +115,43 @@ func startControllers(ctx *ControllerContext) {
 		recorder,
 	).Run(2, ctx.Stop)
 }
+
+func startMetricsCollectionAndServer(ctx *ControllerContext) {
+	machineInformer := ctx.MachineInformerFactory.Machine().V1beta1().Machines()
+	machinesetInformer := ctx.MachineInformerFactory.Machine().V1beta1().MachineSets()
+	machineMetricsCollector := metrics.NewMachineCollector(
+		machineInformer,
+		machinesetInformer,
+		componentNamespace)
+	go func() {
+		if !cache.WaitForCacheSync(ctx.Stop,
+			machineInformer.Informer().HasSynced,
+			machinesetInformer.Informer().HasSynced) {
+			glog.Error("Failed to sync caches")
+			return
+		}
+	}()
+	glog.Info("Synced up machine api informer caches")
```

Comment:
this always log this regardless the result of WaitForCacheSync? unless I'm reading it wrong